### PR TITLE
Accept and verify one Schnorr based signature in block proof

### PR DESCRIPTION
**Pull request by Naviabheeman**
Changed proof to single vector and signed block pubkeys to one aggregate pubkey.
Updated CheckBlockHeader, AbsorbBlockProof and creategenesis block to use schnorr sign and verify
Updated generate, generateToAddress and CombineBlockSigs rpcs to use schnorr signature
Removed signblockthreshold from test framework
Util tests are updated with new genesis block test vectors

### CODE REVIEW

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -117,13 +95,14 @@ static UniValue generatetoaddress(const JSONRPCRequest& request)
             "\nMine blocks immediately to a specified address (before the RPC call returns)\n"
             "\nArguments:\n"
             "1. nblocks      (numeric, required) How many blocks are generated immediately.\n"
-            "2. address      (string, required) The address to send the newly generated coin to.\n"
-            "3. [private keys] (hex string array, required) to sign the generated blocks.\n"
+            "2. address      (string, required) The address to send the newly generated bitcoin to.\n"
+            "3. private key  (hex string array, required) addregate private key to sign the generated blocks.\n"
+            "\nResult:\n"
```
Comment (azuchi) [inline src/rpc/mining.cpp]: `"\nResult:\n"` duplicate

File: src/test/test_tapyrus.cpp [hunk 2]
```diff
@@ -47,7 +47,7 @@ std::ostream& operator<<(std::ostream& os, const uint256& num)
 }
 
 BasicTestingSetup::BasicTestingSetup(const std::string& chainName)
-    : m_path_root(fs::temp_directory_path() / "test_tapyrus" / strprintf("%lu_%i", (unsigned long)GetTime(), (int)(InsecureRandRange(1 << 30)))), signedBlockCondition(getMultisigCondition())
+    : m_path_root(fs::temp_directory_path() / "test_bitcoin" / strprintf("%lu_%i", (unsigned long)GetTime(), (int)(InsecureRandRange(1 << 30))))
```
Comment (azuchi) [inline src/test/test_tapyrus.cpp]: `test_tapyrus` is correct?

File: src/test/test_tapyrus.cpp [hunk 3]
```diff
@@ -57,10 +57,9 @@ BasicTestingSetup::BasicTestingSetup(const std::string& chainName)
     InitSignatureCache();
     InitScriptExecutionCache();
     fCheckBlockIndex = true;
-    assert(signedBlockCondition.threshold == 10);
-    assert(signedBlockCondition.pubkeys.size() == 15);
+    gArgs.SoftSetBoolArg("-server", true);
```
Comment (azuchi) [inline src/test/test_tapyrus.cpp]: What is this `-server` flag?

File: src/test/test_tapyrus.cpp [hunk 4]
```diff
@@ -160,7 +152,7 @@ TestChain100Setup::TestChain100Setup() : TestingSetup(CBaseChainParams::REGTEST)
     {
         std::vector<CMutableTransaction> noTxns;
         CBlock b = CreateAndProcessBlock(noTxns, scriptPubKey);
-        assert(b.proof.size() == Params().GetSignedBlocksCondition().threshold);
+        assert(b.proof.size() == 64);
```
Comment (azuchi) [inline src/test/test_tapyrus.cpp]: Not use `SCHNORR_SIGNATURE_SIZE` constant value?

File: src/test/txindex_tests.cpp [hunk 5]
```diff
@@ -53,7 +53,7 @@ BOOST_FIXTURE_TEST_CASE(txindex_initial_sync, TestChain100Setup)
         CScript coinbase_script_pub_key = GetScriptForDestination(coinbaseKey.GetPubKey().GetID());
         std::vector<CMutableTransaction> no_txns;
         const CBlock& block = CreateAndProcessBlock(no_txns, coinbase_script_pub_key);
-        BOOST_CHECK_EQUAL(block.proof.size(), Params().GetSignedBlocksCondition().threshold);
+        BOOST_CHECK_EQUAL(block.proof.size(), 64);
```
Comment (azuchi) [inline src/test/txindex_tests.cpp]: Not use SCHNORR_SIGNATURE_SIZE constant value?

File: src/test/txvalidationcache_tests.cpp [hunk 6]
```diff
@@ -79,29 +79,29 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_block_doublespend, TestChain100Setup)
         // Test 1: block with both of those transactions should be rejected.
         block = CreateAndProcessBlock(spends, scriptPubKey);
         BOOST_CHECK(chainActive.Tip()->GetBlockHash() != block.GetHash());
-        BOOST_CHECK_EQUAL(block.proof.size(), Params().GetSignedBlocksCondition().threshold);
+        BOOST_CHECK_EQUAL(block.proof.size(), 64);
```
Comment (azuchi) [inline src/test/txvalidationcache_tests.cpp]: Not use SCHNORR_SIGNATURE_SIZE constant value?

File: src/test/validation_block_tests.cpp [hunk 7]
```diff
@@ -174,15 +170,15 @@ BOOST_AUTO_TEST_CASE(processnewblock_signals_ordering)
             bool ignored;
             for (int i = 0; i < 1000; i++) {
                 auto block = blocks[GetRand(blocks.size() - 1)];
-                BOOST_CHECK_EQUAL(block->proof.size(), Params().GetSignedBlocksCondition().threshold);
+                BOOST_CHECK_EQUAL(block->proof.size(), 64);
```
Comment (azuchi) [inline src/test/validation_block_tests.cpp]: Not use SCHNORR_SIGNATURE_SIZE constant value?

File: src/test/test_tapyrus.cpp (hunk 3 above)
Comment (Naviabheeman) [reply src/test/test_tapyrus.cpp]: This is a temporary workaround to allow the test process read "signblockpubkey" argument from command line. In PR#18 "signblockpubkey" argument is removed and this line is also removed.

This is a consequence of this change in src/chainparams.h

``` 
-  CChainParams():aggregatePubkey(GetAggregatePubkey()) {};

+  //require signedblockpubkey argument only in tapyrusd.  
+  CChainParams():aggregatePubkey(gArgs.GetBoolArg("-server", false) ?   GetAggregatePubkeyFromCmdLine() : CPubKey()) {}
```
before this change  "signblockpubkey" was mandatory in `tapyrusd` and `tapyrus-tx`as they  shares the same code.  But in 'tapyrus-tx` this argument is not required.

File: src/test/txindex_tests.cpp (hunk 5 above)
Comment (Naviabheeman) [reply src/test/txindex_tests.cpp]: done.

File: src/test/test_tapyrus.cpp (hunk 2 above)
Comment (Naviabheeman) [reply src/test/test_tapyrus.cpp]: yes, it should be "test_tapyrus".

**Pull request merged**